pkg/chartutil/create.go

```diff
@@ -216,21 +213,50 @@ const defaultIgnore = `# Patterns to ignore when building packages.
 const defaultIngress = `{{- if .Values.ingress.enabled -}}
 {{- $fullName := include "<CHARTNAME>.fullname" . -}}
 {{- $svcPort := .Values.service.port -}}
-{{- if semverCompare ">=1.14-0" .Capabilities.KubeVersion.GitVersion -}}
-apiVersion: networking.k8s.io/v1beta1
+{{- $kubeVersion := .Capabilities.KubeVersion.GitVersion -}}
+{{- if semverCompare ">=1.19-0" $kubeVersion -}}
+apiVersion: networking.k8s.io/v1
 {{- else -}}
-apiVersion: extensions/v1beta1
+apiVersion: networking.k8s.io/v1beta1
```

Comment:
I'd preserve `extensions/v1beta1` support for now.

See here: https://github.com/sagikazarmark/helm-charts/blob/master/charts/tusd/templates/ingress.yaml#L14

Reply:
There are multiple people still running older versions of Kubernetes. Please do not drop API extensions until Helm 4.

Reply:
So proposal would be: 
```
{{- if semverCompare ">=1.19-0" .Capabilities.KubeVersion.GitVersion -}}
apiVersion: networking.k8s.io/v1
{{- else if semverCompare ">=1.14-0" .Capabilities.KubeVersion.GitVersion -}}
apiVersion: networking.k8s.io/v1beta1
{{- else -}}
apiVersion: extensions/v1beta1
{{- end }}
```

Reply:
which probably also needs a change at the ingress creation at the bottom since extensions/v1beta1 does not support a lot of the newer config.

I proposed an additional parameter for helm create to choose a minimal (or target) kubernetes version.
this makes the temple much more readable. there are currently too many if and elses blocks in the template.
you are all experienced helm users - but for new ones it looks pretty scary.

helm is currently build against kubernetes api 0.20.x - which would be theoretically not be compatible to old versions anyway... why not use old versions of helm in combination with old versions of kubernetes?

Reply:
> why not use old versions of helm in combination with old versions of kubernetes?

Because the helm version generating the chart is not necessarily the same as installing the chart and older Helm 3.x versions are compatible with newer charts.